src/api/axiosInstance.js

```diff
@@ -47,6 +47,11 @@ const createRequestInstance = (resolve, timeout) => {
     // Adding conditional because if the server forgets to send back the CORS
     // headers, error.response will be undefined
     const statusCode = error.response ? error.response.status : -1;
+    if (statusCode < 500) {
+      // This request error is to handle problems in the server, so the user
+      // can retry a request. So 404 and 400 errors shouldn't show the error modal
+      return Promise.reject(error);
+    }
```

Comment:
The error handling we have nowadays is not good for some reasons and the main one is that it depends on a promise that the resolve method is stored in the redux, so we can't have API calls with async/await. 

We must have a full refactor in the retry mechanism, so it can handle errors better maybe with the `validateStatus` as optional parameter as @tuliomir suggested.

I will change the code to specify 400 and 404 errors only to be excluded from this error handling, as @r4mmer noticed that the current code has a bug for 429 errors, and then we can proper discuss this new retry mechanism in an [issue](https://github.com/HathorNetwork/hathor-wallet/issues/641).

What do you think?

Note: we have explorer service calls (metadataApi) that don't show the error modal for 404 responses but they use the explorerServiceAxios, not the default axios, which is overridden by the desktop wallet.

Reply:
Code updated: https://github.com/HathorNetwork/hathor-wallet/pull/529/commits/dfdad87afe39d40c5b464282fe16928de0d893e6